driver-core/src/main/com/mongodb/internal/TimeoutContext.java

```diff
@@ -60,7 +76,27 @@ public boolean hasTimeoutMS() {
      * @return true if the timeout has been set and it has expired
      */
     public boolean hasExpired() {
-        return timeout != null && timeout.hasExpired();
+        // Use timeout.remaining instead of timeout.hasExpired that measures in nanoseconds.
```

Comment:
This caught me out - some tests were throwing exceptions but they weren't wrapped because there was less than 1 ms left but more than 1 nano - eek!